Sources/ApodiniNetworking/HTTPServer.swift

```
@@ -0,0 +1,402 @@
+        }
+        let bootstrap = ServerBootstrap(group: eventLoopGroup)
+            .serverChannelOption(ChannelOptions.backlog, value: 256)
+            .serverChannelOption(ChannelOptions.socket(SocketOptionLevel(SOL_SOCKET), SO_REUSEADDR), value: 1)
+            .childChannelInitializer { [unowned self] (channel: Channel) -> EventLoopFuture<Void> in
+                logger.notice("Configuring NIO channel pipeline. TLS: \(tlsConfiguration != nil), HTTP/2: \(enableHTTP2)")
+                if let tlsConfig = tlsConfiguration {
+                    precondition(tlsConfig.applicationProtocols.contains("h2"), "h2 not found in \(tlsConfig.applicationProtocols)")
+                    let sslContext: NIOSSLContext
+                    do {
+                        sslContext = try NIOSSLContext(configuration: tlsConfig)
+                    } catch {
+                        logger.error("Unable to configure TLS: \(error)")
+                        return channel.close(mode: .all)
+                    }
+                    let tlsHandler = NIOSSLServerHandler(context: sslContext)
+                    return channel.pipeline.addHandler(tlsHandler)
+                        .flatMap { () -> EventLoopFuture<Void> in
+                            return channel.configureHTTP2SecureUpgrade { channel in
+                                channel.addApodiniNetworkingHTTP2Handlers(responder: self)
+                            } http1ChannelConfigurator: { channel in
+                                channel.addApodiniNetworkingHTTP1Handlers(responder: self)
+                            }
+                        }
+                        .flatMapError { error in
+                            channel.eventLoop.makeFailedFuture(error)
+                        }
+                } else {
+                    if enableHTTP2 {
+                        // NOTE this doesn't make sense and (probably) doesn't work
```

May this be an outdated comment? Otherwise, why would it not work? Will this be tested? Will this be resolved in this PR? Otherwise I would suggest to track this with an Issue providing a bit more detail about the problem situation.

---

Nope, very much not outdated. Basically no one supports plaintext HTTP/2 (which is what you'd end up configuring in that branch). Technically speaking, plaintext h2 is a thing, but there seems to be a consensus that everyone refuses to implement it (see eg also Nginx and (iirc) Chrome).

[Previously, this would've been a crash](https://github.com/vapor/vapor/blob/a8bd13fa10b14d2c014f002a4eed0558f66bcf6f/Sources/Vapor/HTTP/Server/HTTPServer.swift#L369). If you think that's a more appropriate response we can also add a fatal error here, I'd be up for that.

---

I agree with @lukaskollmer here. HTTP/2 without TLS is not really a thing and I think this should throw an error or even crash the service.